src/twisted/python/_url.py

```diff
@@ -528,9 +528,10 @@ def child(self, *segments):
 
         For example::
 
-            >>> (URL.fromText(u"http://localhost/a/b?x=y")
-                 .child(u"c", u"d").asText())
-            u'http://localhost/a/b/c?x=y'
+            >>> u = URL.fromText(u"http://localhost/a/b?x=y")
+            >>> v = u.child(u"c", u"d")
+            >>> print(v.asText())
+            u'http://localhost/a/b/c/d?x=y'
```

Comment:
@Julian what do you mean by doctests?  There is a builder which generates documentation based on epydoc and reports errors: https://buildbot.twistedmatrix.com/builders/documentation/builds/2533